src/aeso_fcode_to_fate.erl

```diff
@@ -179,63 +183,79 @@ lit_to_fate(L) ->
         {contract_pubkey, K} -> aeb_fate_data:make_contract(K);
         {oracle_pubkey, K}   -> aeb_fate_data:make_oracle(K);
         {oracle_query_id, H} -> aeb_fate_data:make_oracle_query(H);
+        {contract_code, C}   ->
+            FCode = maps:get(C, Env#env.child_contracts),
+            SCode = compile(Env#env.child_contracts, FCode, Env#env.options),
+            ByteCode = aeb_fate_code:serialize(SCode, []),
+            {ok, Version} = aeso_compiler:version(),
+            Code = #{byte_code => ByteCode,
+                     compiler_version => Version,
+                     contract_source => "child_contract_src_placeholder",
```

Comment:
This seems bad. This string is hashed and put on-chain so that people can verify the origin of on-chain contracts. This should be `source_hash` instead, with a hash of something like the full contract source + the name of the child contract.